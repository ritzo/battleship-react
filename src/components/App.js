import React from 'react';
import { Switch, Route, HashRouter } from 'react-router-dom';
import Home from './Home';
import Game from './Game';
import GameOver from './GameOver';

import './App.css';

function App() {
  return (
    <HashRouter>
      <Switch>
        <Route exact path="/" component={Home} />
        <Route path="/game" component={Game} />
        <Route path="/gameOver" component={GameOver} />
      </Switch>
    </HashRouter>
  );
}

export default App;
