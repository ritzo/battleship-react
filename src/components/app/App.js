import React from 'react';
import { Switch, Route, HashRouter } from 'react-router-dom';
import HomeContainer from '../home/HomeContainer';
import GameContainer from '../game/GameContainer';
import GameOverContainer from '../gameOver/GameOverContainer';

import './App.css';

function App() {
  return (
    <HashRouter>
      <Switch>
        <Route exact path="/" component={HomeContainer} />
        <Route path="/game" component={GameContainer} />
        <Route path="/gameOver" component={GameOverContainer} />
      </Switch>
    </HashRouter>
  );
}

export default App;
