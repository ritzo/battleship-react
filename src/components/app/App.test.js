import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('App', () => {
  it('Snapshot', () => {
    const enzymeWrapper = shallow(<App />);

    const component = enzymeWrapper.dive();

    expect(component).toMatchSnapshot();
  });
});
