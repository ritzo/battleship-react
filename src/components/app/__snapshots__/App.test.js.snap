// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Snapshot 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      path="/game"
    />
    <Route
      component={[Function]}
      path="/gameOver"
    />
  </Switch>
</Router>
`;
